components/VM_Arm/configurations/vm.h

```diff
@@ -68,22 +68,28 @@
     attribute int num_extra_frame_caps; \
     attribute int extra_frame_map_address; \
     attribute { \
-        string linux_ram_base; \
-        string linux_ram_paddr_base; \
-        string linux_ram_size; \
-        string linux_ram_offset; \
+        string ram_base; \
+        string ram_paddr_base; \
+        string ram_size; \
+        string ram_offset; \
         string dtb_addr; \
         string initrd_max_size; \
         string initrd_addr; \
-    } linux_address_config; \
+        string kernel_entry_addr = "-1"; \
+    } vm_address_config; \
     attribute { \
-        string linux_name = "linux"; \
+        string kernel_name = "linux"; \
         string dtb_name = "linux-dtb"; \
         string initrd_name = "linux-initrd"; \
-        string linux_bootcmdline = ""; \
-        string linux_stdout = ""; \
+        string kernel_bootcmdline = ""; \
+        string kernel_stdout = ""; \
         string dtb_base_name = ""; \
-    } linux_image_config; \
+        int provide_dtb = true; \
+        int generate_dtb = false; \
+        int provide_initrd = true; \
+        int clean_cache = false; \
+        int map_one_to_one = false; \
```

Comment:
Just a suggestion, could `unity_mapping` be more descriptive? Or `stage2_unity_mapping`?

Reply:
It's called `map_one_to_one`, isn't this clear enough?

Reply:
I think it's not immediately clear this is about MMU.

Reply:
I actually had it as `map_unity` originally, but it was requested that it changed to `map_one_to_one` by @wom-bat.

I feel like its fairly clear, but maybe some documentation about how to use the setting would be beneficial?

Reply:
I think it was @axel-h who requested the one-to-one (https://github.com/seL4/camkes-vm/pull/48#discussion_r1035041892)?

Maybe something like `ram_stage2_direct_translate`? Because it's only being applied to the ram device and the effect is that the stage2 translation is emulating direct-mapping with it's translations...

Reply:
I'm okay with whatever the community decides is best. 

Either way, I'll push up a commit that documents what the flag does.

Just for a simple poll, what do people like best?

- map_unity
- map_one_to_one
- ram_stage2_direct_translate
- stage2_unity_mapping
- ram_direct_translate

Personally, I think referencing the stage 2 mmu gets a bit too complex. How many people even know about how the MMU handles stage 2 mappings?

Reply:
I think it's somehow implicit that this affects the stage 2 translations only. So `ram_direct_translate` or `map_ram_one_to_one`. I have not seen the term "unity" getting used for such translation, so I find this a bit confusing. 
However, I also wonder if a direct mapping is usual intention anyway when the addresses match. So can't we make this the default then? A flag that enables a relaxed mapping might be something to add then - if there is really a need for this.

Reply:
I can live with `map_one_to_one`, not losing my sleep.

Reply:
Thanks, Hannu.